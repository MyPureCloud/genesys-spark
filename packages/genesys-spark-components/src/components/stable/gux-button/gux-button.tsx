import { Component, Element, h, JSX, Prop, State } from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';
import { GuxButtonAccent, GuxButtonType } from './gux-button.types';
import { randomHTMLId } from '@utils/dom/random-html-id';

/**
 * @slot - content
 */

@Component({
  styleUrl: 'gux-button.scss',
  tag: 'gux-button',
  shadow: { delegatesFocus: true }
})
export class GuxButton {
  @Element()
  private root: HTMLElement;
  private buttonId = randomHTMLId('button');

  /**
   * The component button type
   */
  @Prop()
  type: GuxButtonType = 'button';

  /**
   * The component title
   */
  @Prop()
  guxTitle: string;

  /**
   * Indicate if the button is disabled or not
   */
  @Prop()
  disabled: boolean = false;

  @Prop()
  accent: GuxButtonAccent = 'secondary';

  @Prop()
  autofocus: boolean = false;

  @State()
  iconOnly: boolean;

  connectedCallback() {
    this.slotChanged();
  }

  componentWillLoad() {
    trackComponent(this.root, { variant: this.accent });
  }

  render(): JSX.Element {
    return [
      <button
        id={this.buttonId}
        type={this.type}
        disabled={this.disabled}
        class={{
          [`gux-${this.accent}`]: true,
          'gux-icon-only': this.iconOnly
        }}
        aria-label={this.guxTitle}
        autofocus={this.autofocus}
      >
        <slot onSlotchange={this.slotChanged.bind(this)} />
      </button>,
      this.renderTooltip()
    ] as JSX.Element;
  }

  renderTooltip(): JSX.Element {
    return this.guxTitle
      ? ((
          <gux-tooltip-beta for={this.buttonId} visualOnly={true}>
            <div slot="content">{this.guxTitle}</div>
          </gux-tooltip-beta>
        ) as JSX.Element)
      : '';
  }

  private stopEventIfDisabled(event: Event) {
    if (this.disabled) {
      event.stopImmediatePropagation();
      event.stopPropagation();
      event.preventDefault();
    }
  }

  private makeSlotContentDisableable() {
    this.root.shadowRoot.addEventListener('click', (event: MouseEvent) =>
      this.stopEventIfDisabled(event)
    );

    Array.from(this.root.children).forEach(slotElement => {
      slotElement.addEventListener('click', (event: MouseEvent) =>
        this.stopEventIfDisabled(event)
      );
    });
  }

  private hasIconOnly(): boolean {
    const children = Array.from(this.root.children);

    if (children.length === 1) {
      const child = children[0];
      if (child.tagName === 'GUX-ICON') {
        return true;
      }
    } else if (
      children.length === 2 &&
      children[0].tagName === 'GUX-ICON' &&
      ['GUX-TOOLTIP', 'GUX-TOOLTIP-BETA'].includes(children[1].tagName)
    ) {
      return true;
    }

    return false;
  }

  private slotChanged() {
    this.makeSlotContentDisableable();
    this.iconOnly = this.hasIconOnly();
  }
}
