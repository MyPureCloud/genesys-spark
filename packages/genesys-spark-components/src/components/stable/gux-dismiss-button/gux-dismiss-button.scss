gux-button-slot {
  position: absolute;
  top: var(--gse-ui-modal-dismissButton-paddingTop);
  right: var(--gse-ui-modal-dismissButton-paddingRight);

  &.gux-inherit {
    position: inherit;
  }

  .gux-icon-container {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    place-content: stretch center;
    align-items: center;
    width: 8px;

    gux-icon {
      flex: 0 0 auto;
      align-self: auto;
      order: 0;
      width: 16px;
      height: 16px;
      color: var(--gse-ui-dismissButton-foregroundColor);
    }
  }
}
