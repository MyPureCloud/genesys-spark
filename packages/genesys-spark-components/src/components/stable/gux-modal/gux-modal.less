@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/color.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/shadows.less';
@import (reference) '../../../style/zindex.less';

// Style
:host {
  .gux-modal {
    position: fixed;
    inset: 0;
    z-index: var(--gux-zindex-modal, @gux-zindex-modal);
    display: flex;
    align-items: center;
    justify-content: center;
    color: @gux-black-50;
    background: fade(@gux-black-30, 80%);

    .gux-modal-container {
      .gux-shadow-30();

      position: relative;
      box-sizing: border-box;
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      padding: 40px 0;
      background: @gux-grey-100;
      border: 1px solid @gux-grey-30;
      border-radius: 4px;

      &.gux-small {
        width: 416px;
        max-height: e(%('min(368px, 100vh - 2 * %s)', @gux-spacing-large));
      }

      &.gux-medium {
        width: 672px;
        max-height: e(%('min(640px, 100vh - 2 * %s)', @gux-spacing-large));
      }

      &.gux-large {
        width: 940px;
        max-height: e(%('min(640px, 100vh - 2 * %s)', @gux-spacing-large));
      }

      &.gux-dynamic {
        max-width: calc(100vw - 2 * @gux-spacing-large);
        max-height: calc(100vh - 2 * @gux-spacing-large);

        .gux-modal-content {
          max-height: none;
        }
      }

      .gux-modal-header {
        padding: 0 40px 24px;
        margin: 0;
        .h1();
      }

      .gux-modal-content {
        max-height: 432px;
        padding: 0 40px;
        margin: 0 8px 24px;
        overflow-y: auto;

        &.gux-no-buttons {
          margin-bottom: 0;
        }
      }

      .gux-button-footer {
        display: flex;
        justify-content: space-between;
        padding: 0 40px;

        .gux-left-align-buttons {
          gux-button {
            padding-right: 5px;
          }
        }

        .gux-right-align-buttons {
          gux-button {
            padding-left: 5px;
          }
        }
      }
    }

    .gux-modal-container {
      @media (max-width: 416px) {
        &.gux-small,
        &.gux-medium,
        &.gux-large {
          width: 100%;
          height: 100%;
        }
      }
    }
  }
}
