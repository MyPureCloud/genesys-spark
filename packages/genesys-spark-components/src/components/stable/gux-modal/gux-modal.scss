@use '../../../style/typography.scss';

::slotted(button:active:enabled) {
  color: var(--gse-ui-link-active-foregroundColor);
  text-decoration: underline;
  background: none;
}

slot[name='start-align-buttons']::slotted(:not(button, gux-button-slot)) {
  display: flex;
  flex-direction: row;
  gap: var(--gse-ui-modal-buttonBar-gap);
  align-content: flex-start;
  align-items: center;
}

:host {
  dialog {
    box-sizing: border-box;
    justify-content: space-between;
    border: 1px solid #b4bccb; // TODO: COMUI-2300
    border-radius: var(--gse-ui-modal-borderRadius);
    box-shadow: var(--gse-ui-modal-boxShadow-x) var(--gse-ui-modal-boxShadow-y)
      var(--gse-ui-modal-boxShadow-blur) var(--gse-ui-modal-boxShadow-spread)
      var(--gse-ui-modal-boxShadow-color);

    &::backdrop {
      // token isn't working?
      // background: var(--gse-ui-modal-shroudColor);
      background: #4d6182cc;
    }

    .gux-modal-container {
      display: flex;
      flex-direction: column;
      padding: var(--gse-ui-modal-padding);

      &.gux-small {
        width: var(--gse-ui-modal-small-width);
        max-height: min(368px, 100vh - 2 * #{24px});
      }

      &.gux-medium {
        width: var(--gse-ui-modal-medium-width);
        max-height: min(640px, 100vh - 2 * #{24px});
      }

      &.gux-large {
        width: var(--gse-ui-modal-large-width);
        max-height: min(640px, 100vh - 2 * #{24px});
      }

      &.gux-dynamic {
        max-width: calc(100vw - 2 * 24px);
        max-height: calc(100vh - 2 * 24px);

        .gux-modal-content {
          max-height: none;
        }
      }

      .gux-modal-header {
        margin: 0;
        font-family: var(--gse-ui-modal-heading-fontFamily);
        font-size: var(--gse-ui-modal-heading-fontSize);
        font-weight: var(--gse-ui-modal-heading-fontWeight);
        line-height: var(--gse-ui-modal-heading-fontFamily);
        color: var(--gse-ui-modal-headerColor);
        letter-spacing: var(--gse-ui-modal-heading-letterSpacing);

        @include typography.heading-xl;
      }

      .gux-modal-content {
        max-height: 432px;
        margin-top: var(--gse-ui-modal-gap);
        margin-bottom: var(--gse-ui-modal-gap);
        overflow-y: auto;

        &.gux-no-buttons {
          margin-bottom: 0;
        }
      }

      .gux-button-footer {
        display: flex;
        justify-content: space-between;
      }
    }

    .gux-modal-container {
      @media (max-width: 416px) {
        &.gux-small,
        &.gux-medium,
        &.gux-large {
          width: 100%;
          height: 100%;
        }
      }
    }
  }
}
