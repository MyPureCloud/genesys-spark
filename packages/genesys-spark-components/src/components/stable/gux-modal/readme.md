# gux-modal

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                                                                                                                                                                                                                                                                                                                          | Type                                          | Default     |
| -------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------- | ----------- |
| `open`   | `open`    | Indicates/sets whether or not the modal is open. On a native dialog, you should not toggle the open attribute, due to the unusual behaviors described [here](https://html.spec.whatwg.org/multipage/interactive-elements.html#attr-dialog-open) In this component, it is safe as this property acts as a proxy for calls to `showModal` and `close`. | `boolean`                                     | `false`     |
| `size`   | `size`    | Indicates the size of the modal (small, medium or large)                                                                                                                                                                                                                                                                                             | `"dynamic" \| "large" \| "medium" \| "small"` | `'dynamic'` |


## Events

| Event        | Description                           | Type                |
| ------------ | ------------------------------------- | ------------------- |
| `guxdismiss` | Fired when a user dismisses the modal | `CustomEvent<void>` |


## Methods

### `close() => Promise<void>`



#### Returns

Type: `Promise<void>`



### `showModal() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Dependencies

### Depends on

- [gux-dismiss-button](../gux-dismiss-button)

### Graph
```mermaid
graph TD;
  gux-modal --> gux-dismiss-button
  gux-dismiss-button --> gux-button-slot
  gux-dismiss-button --> gux-icon
  style gux-modal fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
