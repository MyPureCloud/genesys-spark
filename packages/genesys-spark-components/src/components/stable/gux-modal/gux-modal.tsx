import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop,
  Method,
  Watch
} from '@stencil/core';

import { randomHTMLId } from '@utils/dom/random-html-id';
import { trackComponent } from '@utils/tracking/usage';

import { GuxModalSize } from './gux-modal.types';

@Component({
  styleUrl: 'gux-modal.scss',
  tag: 'gux-modal',
  shadow: { delegatesFocus: true }
})
export class GuxModal {
  private dialogElement: HTMLDialogElement;

  @Element()
  private root: HTMLElement;

  /**
   * Indicates the size of the modal (small, medium or large)
   */
  @Prop()
  size: GuxModalSize = 'dynamic';

  /**
   * Indicates/sets whether or not the modal is open. On a native dialog, you should not toggle the
   * open attribute, due to the unusual behaviors described [here](https://html.spec.whatwg.org/multipage/interactive-elements.html#attr-dialog-open)
   * In this component, it is safe as this property acts as a proxy for calls to `showModal` and `close`.
   */
  @Prop({ mutable: true })
  open: boolean = false;

  /**
   * Fired when a user dismisses the modal
   */
  @Event()
  guxdismiss: EventEmitter<void>;

  /**
   * "Renders" the open state of the modal
   */
  @Watch('open')
  private syncOpenState() {
    if (this.open) {
      this.dialogElement.showModal();
    } else {
      this.dialogElement.close();
    }
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async showModal(): Promise<void> {
    this.open = true;
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async close(): Promise<void> {
    this.open = false;
  }

  componentWillLoad(): void {
    trackComponent(this.root, { variant: `${this.size}` });
  }

  componentDidLoad(): void {
    this.syncOpenState();
  }

  private hasModalTitleSlot(): boolean {
    return Boolean(this.root.querySelector('[slot="title"]'));
  }

  private hasFooterButtons(): boolean {
    return (
      Boolean(this.root.querySelector('[slot="start-align-buttons"]')) ||
      Boolean(this.root.querySelector('[slot="end-align-buttons"]'))
    );
  }

  render(): JSX.Element {
    const hasModalTitleSlot = this.hasModalTitleSlot();
    const hasFooterButtons = this.hasFooterButtons();
    const titleID: string = randomHTMLId();

    return (
      <dialog
        onClose={this.onCloseHandler.bind(this)}
        ref={el => (this.dialogElement = el)}
        aria-labelledby={hasModalTitleSlot ? titleID : null}
      >
        <div class={`gux-modal-container gux-${this.size}`}>
          <gux-dismiss-button
            onClick={this.onDismissHandler.bind(this)}
          ></gux-dismiss-button>

          {hasModalTitleSlot && this.renderTitle(titleID)}

          <div
            class={{
              'gux-modal-content': true,
              'gux-no-buttons': !hasFooterButtons
            }}
          >
            <p>
              <slot name="content" />
            </p>
          </div>
          {hasFooterButtons && this.renderButtonFooter()}
        </div>
      </dialog>
    ) as JSX.Element;
  }

  private renderTitle(titleID: string): JSX.Element {
    return (
      <h1 class="gux-modal-header" id={titleID}>
        <slot name="title" />
      </h1>
    ) as JSX.Element;
  }

  private renderButtonFooter(): JSX.Element {
    return (
      <div class="gux-button-footer">
        <div class="gux-start-align-buttons">
          <slot name="start-align-buttons" />
        </div>

        <div class="gux-end-align-buttons">
          <slot name="end-align-buttons" />
        </div>
      </div>
    ) as JSX.Element;
  }
  private onCloseHandler(): void {
    this.guxdismiss.emit();
  }

  private onDismissHandler(): void {
    this.dialogElement.close();
  }
}
