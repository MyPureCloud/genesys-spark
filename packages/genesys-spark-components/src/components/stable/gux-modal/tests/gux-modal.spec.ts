import { newSpecPage } from '@test/specTestUtils';
import { GuxButton } from '../../gux-button/gux-button';
import { GuxModal } from '../gux-modal';
import { GuxDismissButton } from '../../gux-dismiss-button/gux-dismiss-button';
import { MockHTMLElement } from '@stencil/core/mock-doc';
import { SpecPage } from '@stencil/core/internal';

const components = [GuxButton, GuxModal, GuxDismissButton];
const language = 'en';

//Mocks
const showModal = jest.fn();
const close = jest.fn();

describe('gux-modal', () => {
  beforeAll(() => {
    // Required until JSDOM supports the dialog element. See:
    // https://github.com/jsdom/jsdom/issues/3294
    // https://github.com/jsdom/jsdom/pull/3403
    Object.assign(MockHTMLElement.prototype, {
      showModal: showModal,
      close: close
    });
  });

  beforeEach(() => {
    showModal.mockReset();
    close.mockReset();
  });

  it('Should open when rendered with an open attribute', async () => {
    await modalPage(openModal);

    expect(showModal).toHaveBeenCalled();
  });

  it('Should not open when rendered without an open attribute', async () => {
    await modalPage(closedModal);

    expect(showModal).not.toHaveBeenCalled();
  });

  it('Should close when the open property is set to false', async () => {
    const page = await modalPage(openModal);

    page.rootInstance.open = false;
    expect(close).toHaveBeenCalled();
  });

  it('Should open when the open property is set to true', async () => {
    const page = await modalPage(closedModal);

    page.rootInstance.open = true;
    expect(showModal).toHaveBeenCalled();
  });

  it('Should keep the open property in sync when opened/closed via method', async () => {
    const page = await modalPage(closedModal);
    const modal = page.rootInstance;

    expect(modal.open).toBe(false);

    modal.showModal();
    expect(modal.open).toBe(true);

    modal.close();
    expect(modal.open).toBe(false);
  });

  describe('#render', () => {
    [
      {
        description: 'should render small modal',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render medium modal',
        html: `
          <gux-modal size="medium">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render large modal',
        html: `
          <gux-modal size="large">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without a title',
        html: `
          <gux-modal size="large">
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal without buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just left align buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render modal with just right align buttons',
        html: `
          <gux-modal size="small">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description:
          'should render modal with a specified initial focus element',
        html: `
          <gux-modal initial-focus="#cancelButton">
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button id="cancelButton">Cancel</gux-button>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      },
      {
        description: 'should render small modal by default',
        html: `
          <gux-modal>
            <div slot="title">Modal Title</div>
            <div slot="content">This contains the modal content.</div>
            <div slot="start-align-buttons">
                <gux-button-slot><button>Cancel</button></gux-button-slot>
            </div>
            <div slot="end-align-buttons">
              <gux-button-slot accent='primary'><button>Accept</button></gux-button-slot>
            </div>
          </gux-modal>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxModal);

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});

const openModal = `
  <gux-modal size="small" open>
    <div slot="content">This contains the modal content.</div>
  </gux-modal>`;

const closedModal = `
  <gux-modal size="small">
    <div slot="content">This contains the modal content.</div>
  </gux-modal>`;

async function modalPage(html: string): Promise<SpecPage> {
  return await newSpecPage({ components, html, language });
}
