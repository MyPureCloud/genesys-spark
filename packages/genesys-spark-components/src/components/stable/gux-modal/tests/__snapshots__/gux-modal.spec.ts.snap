// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-modal #render should render large modal 1`] = `
<gux-modal size="large">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-large gux-modal-container">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render medium modal 1`] = `
<gux-modal size="medium">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-medium gux-modal-container">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with a specified initial focus element 1`] = `
<gux-modal initial-focus="#cancelButton">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-dynamic gux-modal-container">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button id="cancelButton">
      <mock:shadow-root>
        <button class="gux-secondary" type="button">
          <slot></slot>
        </button>
      </mock:shadow-root>
      Cancel
    </gux-button>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just left align buttons 1`] = `
<gux-modal size="small">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal with just right align buttons 1`] = `
<gux-modal size="small">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without a title 1`] = `
<gux-modal size="large">
  <mock:shadow-root>
    <dialog>
      <div class="gux-large gux-modal-container">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render modal without buttons 1`] = `
<gux-modal size="small">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content gux-no-buttons">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal 1`] = `
<gux-modal size="small">
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-modal-container gux-small">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;

exports[`gux-modal #render should render small modal by default 1`] = `
<gux-modal>
  <mock:shadow-root>
    <dialog aria-labelledby="gux-i">
      <div class="gux-dynamic gux-modal-container">
        <gux-dismiss-button>
          <mock:shadow-root>
            <gux-button-slot accent="ghost">
              <button title="Dismiss" type="button">
                <div class="gux-icon-container">
                  <gux-icon decorative="" icon-name="close"></gux-icon>
                </div>
              </button>
            </gux-button-slot>
          </mock:shadow-root>
        </gux-dismiss-button>
        <h1 class="gux-modal-header" id="gux-i">
          <slot name="title"></slot>
        </h1>
        <div class="gux-modal-content">
          <p>
            <slot name="content"></slot>
          </p>
        </div>
        <div class="gux-button-footer">
          <div class="gux-start-align-buttons">
            <slot name="start-align-buttons"></slot>
          </div>
          <div class="gux-end-align-buttons">
            <slot name="end-align-buttons"></slot>
          </div>
        </div>
      </div>
    </dialog>
  </mock:shadow-root>
  <div slot="title">
    Modal Title
  </div>
  <div slot="content">
    This contains the modal content.
  </div>
  <div slot="start-align-buttons">
    <gux-button-slot>
      <button>
        Cancel
      </button>
    </gux-button-slot>
  </div>
  <div slot="end-align-buttons">
    <gux-button-slot accent="primary">
      <button>
        Accept
      </button>
    </gux-button-slot>
  </div>
</gux-modal>
`;
